### ISSUE TITLE
Change API call signatures to use ErrorType instead of NSError

### ISSUE THREAD

**Issue by cojoj** [czechboy0/XcodeServerSDK#81]
Right now most of our API endpoint methods (these located in `XcodeServer` `extension`s) are constructed in _non-swifty_ way - they return optional `NSError` pointers.

How about gradually changing them to `throw`ing methods? 
There are some benefits of doing this:
1. **Swift**y way (most important argument... 😜)
2. Reduce amount of `Optional`s
3. More descriptive errors, as they'll be `enum`s

I know, I know, I may be overreacting... Maybe that's because I have too much free time? 😝

**Comment by czechboy0** [comment]
Can you provide an example? AFAIK, all our APIs are _asynchronous_, so the callback block is called much later after the call of the API method. Throw only works with _synchronous_ calls.

What we **could** do is change the signature from `NSError` to `ErrorType`. That would be a good improvement, because it would allow for error enum types.

**Comment by cojoj** [comment]
Right, _async_... And I'm back home 😉 But we have more methods which may be considered this way eg. `createRequest()` returns `NSMutableRequest?`.

Anyway, `ErrorTypes` sounds great!

**Comment by czechboy0** [comment]
Yup, `createRequest()` would be a good candidate for `throw`ing.

**Comment by czechboy0** [comment]
Action: **look into changing the error type in API call block callbacks from `NSError` to `ErrorType`.**
